frontend/src/utilities/useWatchOperatorSubscriptionStatus.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { fetchOperatorSubscriptionStatus } from '~/services/operatorSubscriptionStatusService';
+import { SubscriptionStatusData } from '~/types';
+import { useDeepCompareMemoize } from './useDeepCompareMemoize';
+import { POLL_INTERVAL } from './const';
+
+export const useWatchOperatorSubscriptionStatus = (): SubscriptionStatusData | null => {
```

Comment:
out of curiosity, why didn't you use `useFetchState` -- it comes with a lot of the same functionality for near free